### DISCUSSION TITLE
devshell

### DISCUSSION THREAD

**Discussion by mightyiam** [90-008/nix-cargo-integration#164] [Show and tell]
Here's what I'm coming up with for a devshell, and it still doesn't have `cc` for some reason:

```nix
  perSystem =
    { config, pkgs, ... }:
    {
      devshells.default = {
        devshell = {
          packages = [ pkgs.rust-analyzer ];
          packagesFrom =
            removeAttrs config.nci.outputs [ "default" ] |> lib.mapAttrsToList (_: output: output.devShell);
        };
        env = lib.concatMapAttrs (_: crate: crate.drvConfig.env) config.nci.crates |> lib.attrsToList;
      };
    };
```

I'd love to just use a value or two that nci would provide, instead.

**Comment by 90-008** [comment]
are you trying to use a numtide devshell? i just merged a pr that adds support for it, you can check it in `examples/numtide-devshell`.

**Reply by mightyiam** [reply]
I am, actually! I'll try it out!

**Reply by mightyiam** [reply]
That is dope! Thank you!
